src/clarity-angular/data/datagrid/datagrid-items-trackby.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+    Directive,
+    Input,
+    TrackByFunction
+} from "@angular/core";
+
+import {Items} from "./providers/items";
+
+@Directive({
+    selector: "[ngForTrackBy]",
+})
+export class DatagridItemsTrackBy {
+
+    constructor(private _items: Items) {}
```

Comment:
This breaks for breaks for `ngForTrackBy` outside of a datagrid. The Items provider needs to be optional.